core/dbt/clients/jinja.py

```diff
@@ -52,8 +52,13 @@ def _compile(self, source, filename):
             )
             # encode, though I don't think this matters
             filename = jinja2._compat.encode_filename(filename)
-            # put ourselves in the cache using the 'lazycache' method
-            linecache.cache[filename] = (lambda: source,)
+            # put ourselves in the cache
```

Comment:
This is internal python debugger magic, it injects the source code we generate via jinja into the `linecache`, which is how pdb and its derivatives know how to show you the source code as you step through code. Basically it means you aren't flying blind while stepping through dbt code in a debugger.

The old version that used the 'lazycache' was apparently a python3-only feature. I switched it to this because before this, `import pdb` caused python 2.7 to crash and I needed to debug some python 2-only issues.